stix_shifter_modules/infoblox/stix_translation/query_constructor.py

```diff
@@ -0,0 +1,434 @@
+import datetime
+import json
+import logging
+import re
+import time
+
+from stix_shifter_utils.stix_translation.src.json_to_stix import observable
+from stix_shifter_utils.stix_translation.src.patterns.pattern_objects import (
+    ObservationExpression, ComparisonExpression,
+    ComparisonExpressionOperators, ComparisonComparators, Pattern,
+    CombinedComparisonExpression, CombinedObservationExpression, ObservationOperators,
+    StartStopQualifier
+)
+from .transformers import InfobloxToDomainName, TimestampToSeconds
+
+REFERENCE_DATA_TYPES = {
+    "qip": ["ipv4", "ipv4_cidr"],
+    "value": ["ipv4", "ipv4_cidr", "domain_name"],
+    "qname": ["domain_name"],
+    "ip": ["ipv4", "ipv4_cidr", "ipv6", "ipv6_cidr"]
+}
+REFERENCE_FIELDS = ('src_ref.value', 'hostname_ref.value',
+    'ip_ref.value', 'extensions.dns-ext.question.domain_ref.value'
+)
+
+START_STOP_STIX_QUALIFIER = r"START((t'\d{4}(-\d{2}){2}T\d{2}(:\d{2}){2}(\.\d+)?Z')|(\s\d{13}\s))STOP"
+TIMESTAMP_MILLISECONDS = r"\.\d+Z$"
+
+THREAT_LEVEL_MAPPING = {
+    "HIGH": 3,
+    "MEDIUM": 2,
+    "LOW": 1
+}
+
+logger = logging.getLogger(__name__)
+
+class DuplicateFieldException(Exception):
+    pass
+
+class QueryStringPatternTranslator:
+    comparator_lookup = {
+        'tideDbData': {
+            ComparisonExpressionOperators.And: "&",
+            ObservationOperators.And: '&',
+            ComparisonExpressionOperators.Or: "&",
+            ObservationOperators.Or: '&',
+            ComparisonComparators.Equal: "=",
+            ComparisonComparators.GreaterThan: "=",
```

Comment:
@grimmjow8 I know this has already been merged, but we've been doing some work around the operators and noticed that you are assigning `=` to STIX operators that aren't equal. I don't fully understand how the query syntax for Infoblox works, but wouldn't this give a false result when using this in a query? It also looks like Infoblox only supports an AND operator, not OR. In cases where an operator is not supported, it should just be omitted from the comparator lookup.

Reply:
Hey @delliott90! For the most part, it should give accurate results. Non-equal operators, like `>`,`>=`, etc are only supported by a few fields. Attempts to do that on non-capable fields should result in errors. For the fields that do support it, infoxblox api has particular field formats that are used. So for `imported` field, supported query parameters are `imported_from_date` and `imported_to_date`. Probably not the most intuitive out of the box but is still applicable.

In regards to `AND`/`OR`, we initially excluded it but it was recommended on this PR that a workaround could be to split it into multiple queries. We went with that since it seemed to provide accurate results.